### PR TITLE
WindowBase|WindowSDL: Add drop position for all on_drop_xxx events

### PR DESCRIPTION
**Pull request by pythonic64**
Adds dispatching of the drop position (in coords relative to the window left/top) with all `on_drop_xxx` events. Documentation about the events is updated. This is a follow-up on https://github.com/kivy/kivy/pull/7786.

Maintainer merge checklist:
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [x] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

File: kivy/core/window/window_sdl2.py
Comment (matham) [inline kivy/core/window/window_sdl2.py:778]: I think we should bake in to the docs that the default if it's not in the window or if it's not a mouse but e.g. a touch, that it will be `-1` rather than `0`!?

Comment (pythonic64) [reply kivy/core/window/window_sdl2.py:778]: Negative values are valid as they can be used to test if the drop occurred on the title bar or on the left window edget. Docs do state that `x` and `y` are the cursor positions at the time of the drop and SDL does not send those values with the `dropfile` event, only `filename` is sent so we don't know from where the drop originated.

**Review by matham — APPROVED** [review]

**Comment by pythonic64** [comment]
Maybe we need more in the doc for case when/if the drop is not coming from the mouse.

**Comment by matham** [comment]
>Maybe we need more in the doc for case when/if the drop is not coming from the mouse?

Possibly. But we can always update it later if we figure out it's possible to get a drop without the mouse!? It's up to you.

**Comment by akshayaurora** [comment]
looks good,  LGTM!

**Comment by matham** [comment]
@akshayaurora please make sure to squash before merging, especially if there are multiple commits that are not separate features.

@pythonic64 it may be a good idea to mark PRs as WIP until you feel it's ready to merge so we don't merge before you're ready if you're still considering changes. We can still review it before that though. There's a button that makes it a draft to block merging until you're ready!

**Comment by pythonic64** [comment]
@matham At the time of the creation the request looked ready to me, but now I think we should have added `*args` to those events, just in case that SDL might pass a drop origin.

**Pull request merged**
